google/cloud/storage/client.h

```diff
@@ -1675,6 +1675,7 @@ class Client {
   template <typename... Options>
   StatusOr<std::vector<BucketAccessControl>> ListBucketAcl(
       std::string const& bucket_name, Options&&... options) {
+    auto const span = MakeSpan(std::forward<Options>(options)...);
```

Comment:
For the record, I'm a little concerned about returning `OptionSpan`s.  It seems to depend on copy elision to do the right thing, but I'm still thinking it through.  Remember that the `OptionSpan` ctor and dtor write to a (thread) global, and its set/restore operations need to be FIFO.  Perhaps `OptionSpan` should not be copyable/movable.

A safer tactic would be to have the pack consumer just return an `Options`, and then have these functions instantiate the `OptionsSpan` directly.

Reply:
Fair enough.  I would prefer to merge the current PRs and then make one PR fixing this.  Is that Okay with you?  Otherwise I will pull them and create one large PR, redoing the work with a few PRs already merged is more convoluted than I wish.



nit: I think you meant LIFO.

Reply:
> Fair enough. I would prefer to merge the current PRs and then make one PR fixing this. Is that Okay with you?

Of course.  Whatever is easiest.

> nit: I think you meant LIFO.

Indeed.